### DISCUSSION TITLE
Better app search filter

### DISCUSSION THREAD

**Discussion by nico7master** [tanujnotes/Olauncher#226] [Ideas]
I would like to request a change, but I am new to kotlin and this process in general so would be amazing if someone could take my pseudo code and put it into actual code ;)
I would like to change the filter when you search and type the name of the app.
The current implementation is that it looks if any app name contains the typed chars. But that doesn't make sense most of the time since you would never look up an app from the middle of the name and it creates to many false positives imo.
So a better way would be the startwith function, looking only from the start of the name. But a app name could have multiple words in one so you would first create an array of the words separated by a space and the look though each of them and see if they start with your typed chars. 
Hope that makes sense and someone is willing to make the change. 
I really like the launcher but this often annoyed me since sometimes there is only one app starting with those 2 letters but it gives 6 because those letters also are contained.
Thanks!

**Comment by tanujnotes** [comment]
The current implementation is intentional because it allows users to develop their own shortcuts to quickly open apps. For example, I type pp to open WhatsApp, tt to open Twitter, gr for Instagram, etc.

**Reply by nico7master** [reply]
But everybody else who not uses any shortcuts is way worse off. 
I think with 3 letters you have a 99% chance of actually getting the right app and its not slower to type an extra letter instead of a weird combination which you found out that works. 
I think this is a feature only useful to anybody who memorizes special characters for special names which almost nobody does I bet. And then you download another app which also has pp in the name and you have to find something else. Doesn't make really sense to me.

Do you understand my problem?

**Reply by nico7master** [reply]
I am also not memorizing shortcuts for all of my apps. And I am using most of my apps downloaded regularly.

**Reply by tanujnotes** [reply]
I've interacted with thousands of Olauncher users at this point. If this was an issue, trust me, I'd know. There have been no changes in Olauncher app search in about last 2 years and while there have been many complaints, most of the users are happy about how the search works. For now, there's no plan to make any changes in it.

**Reply by nico7master** [reply]
Its not an issue it still works. But would just work better with looking from the start. I understand you like it like that. Too bad. 
Thanks for replying though!
